Image search

mikechao opened the pull request:
Adds the ability to do an image search by using Brave Search Image Search API.

Includes typescript interfaces that corresponds to image search response object
https://api-dashboard.search.brave.com/app/documentation/image-search/responses

One thing I am not sure about is the Query Parameters for Image search. (https://api-dashboard.search.brave.com/app/documentation/image-search/query) It seems to be subset of the Web Search query parameters. I tested locally the imageSearch method with the  web query param of 'freshness' which is not in the image query parameters. It is just ignored. It might create confusion for users that use IDE to suggest/autocomplete parameters.

mikechao committed:
Added interface for Image search response

mikechao committed:
Added imageSearch api

mikechao committed:
Updated README

erik-balfe commented:
Hi! Great job.  

For the image query params, I’d suggest creating an interface that extends from the main query params (`BraveSearchOptions`) in `types.ts`. Here’s how you could do it:  

```ts
export interface ImageSearchOptions extends Pick<BraveSearchOptions, 'country' | 'search_lang' | 'safesearch' | 'spellcheck' | 'count'> {
  q: string;
}
```  

Then, you can use this interface in your code for the image search query params.

mikechao committed:
Added ImageSearchOptions using Pick

mikechao commented:
Hello,

I added the ImageSearchOptions as you suggested and incorporated it as part of the imageSearch method. I did not add q: string as part of ImageSearchOptions because it is already a parameter of the imageSearch method. 

Still learning TypeScript, thanks for letting know me about Pick.

Any other suggestions?
